subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformer.java

```diff
@@ -290,9 +300,14 @@ public void visitInputProperty(String propertyName, PropertyValue value, boolean
 
             @Override
             public void visitOutputFileProperty(String propertyName, boolean optional, PropertyValue value, OutputFilePropertyType filePropertyType) {
-                validationContext.visitPropertyProblem(WARNING,
-                    propertyName,
-                    "is annotated with an output annotation"
+                validationContext.visitPropertyProblem(problem ->
```

Comment:
Didn't we say this never happens? So either we need to add a test for checking this message or we throw an exception from this method.